lib/t/matches.ex

```diff
@@ -302,20 +314,37 @@ defmodule T.Matches do
 
   @spec list_matches(uuid) :: [%Match{}]
   def list_matches(user_id) do
-    Match
-    |> where([m], m.user_id_1 == ^user_id or m.user_id_2 == ^user_id)
-    |> where([m], m.id not in subquery(archived_match_ids_q(user_id)))
+    matches_with_undying_events_q()
+    |> where([match: m], m.user_id_1 == ^user_id or m.user_id_2 == ^user_id)
+    |> where([match: m], m.id not in subquery(archived_match_ids_q(user_id)))
     |> order_by(desc: :inserted_at)
-    |> join(:left, [m], t in Timeslot, on: m.id == t.match_id)
-    |> join(:left, [m, t], c in MatchContact, on: m.id == c.match_id)
-    |> select([m, t, c], {m, t, c})
+    |> join(:left, [match: m], t in Timeslot, as: :t, on: m.id == t.match_id)
+    |> join(:left, [match: m], c in MatchContact, as: :c, on: m.id == c.match_id)
+    |> select([match: m, undying_event: e, t: t, c: c], {m, t, c, e.timestamp})
     |> Repo.all()
-    |> Enum.map(fn {match, timeslot, contact} ->
-      %Match{match | timeslot: timeslot, contact: contact}
+    |> Enum.map(fn {match, timeslot, contact, undying_event_timestamp} ->
+      expiration_date =
+        unless undying_event_timestamp do
+          expiration_date(match)
+        end
+
+      %Match{
+        match
+        | timeslot: timeslot,
+          contact: contact,
```

Comment:
does %Match{} need `timeslot` & `contact` when it has interaction?

Reply:
why not?

Reply:
no reason
rather wanted to understand rationale - maybe you had something in mind where they will be needed later or something.